paddle/fluid/platform/mkldnn_reuse.h

```diff
@@ -35,7 +35,8 @@ using user_function = std::function<std::shared_ptr<float>(const float*)>;
 using memory = mkldnn::memory;
 
 template <typename T, typename TForward,
-          typename TBackward = mkldnn_dummy_primitive>
+          typename TBackward = mkldnn_dummy_primitive,
+          typename TBackward_params = mkldnn_dummy_primitive>
```

Comment:
If this is only Convolution specific then wouldn't it be more readable to have it in its own handler class? Moreover what if we would like to reuse this class for other operations (like recurrent RNN, GRU, LSTM) which may require even more additional parameter types?

Reply:
It is not only convolution specific. It is specific to all primitives that are having trainable weights e.g. conv, conv_transpose, fully connected, prelu...  As for LSTM (when we get to training) I'm not sure perhaps it would make sense to do some separate class